redisvl/schema/schema.py

```diff
@@ -0,0 +1,306 @@
+        fields = kwargs.pop("fields", {})
+        return cls.from_params(**{
+            "name": name,
+            "prefix": prefix,
+            "storage_type": storage_type,
+            "key_separator": key_separator,
+            "fields": fields,
+            **kwargs
+        })
+
+    # @classmethod
+    # def from_sample(cls, sample: Dict[str, Any]) -> None:
+    #     """Construct a Schema from a sample of data."""
+    #     generator = SchemaGenerator()
+    #     schema = generator.generate(metadata=sample, strict=True)
+    #     return cls(index = schema.index, fields = schema.fields)
+    # TODO can't use something like this until we have extended the schema generator
+
+    @property
+    def index_name(self) -> str:
+        return self._index.name
+
+    @property
+    def index_prefix(self) -> str:
+        return self._index.prefix
+
+    @property
+    def key_separator(self) -> str:
+        return self._index.key_separator
+
+    @property
+    def storage_type(self) -> str:
+        return self._index.storage_type
+
+    @property
+    def index_fields(self):
+        redis_fields = []
+        for field_name in self._fields.__fields__.keys():
+            field_group = getattr(self._fields, field_name)
+            if field_group is not None:
+                for field in field_group:
+                    redis_fields.append(field.as_field())
+        return redis_fields
+
+    def dump(self) -> Dict[str, Any]:
+        """
+        Dump the RedisVL schema to a dictionary.
+
+        Returns:
+            The RedisVL schema as a dictionary.
+        """
+        return {
+            "index": self._index.dict(),
+            "fields": self._fields.dict()
+        }
+
+    def write_to_yaml(self, path: str) -> None:
+        """
+        Write the schema to a yaml file.
+
+        Args:
+            path (str): The yaml file path where the schema will be written.
+
+        Raises:
+            TypeError: If the provided file path is not a valid YAML file.
+        """
+        if not path.endswith(".yaml"):
+            raise TypeError("Invalid file path. Must be a YAML file.")
+
+        schema = self.dump()
+        with open(path, "w") as f:
+            yaml.dump(schema, f)
+
+
+def read_schema(file_path: str) -> Schema:
+    """
+    Create a Schema instance from a YAML file.
+    Args:
+        file_path: The path to the YAML file.
+    Returns:
+        A Schema instance.
+    Raises:
+        ValueError: If the file path is not a YAML file.
+        FileNotFoundError: If the YAML file does not exist.
+    """
+    if not file_path.endswith(".yaml"):
+        raise ValueError("Must provide a valid YAML file path")
+
+    fp = Path(file_path).resolve()
+    if not fp.exists():
+        raise FileNotFoundError(f"Schema file {file_path} does not exist")
+
+    with open(fp, "r") as f:
+        schema = yaml.safe_load(f)
+
+    return Schema(**schema)
+
+
+class SchemaGenerator:
```

Comment:
I think this should be a classmethod that returns a ``Schema`` object. Any reason this should be a long lived object?

Reply:
I agree. Was waiting to do this before we had a change to get some more alignment. Will do today